src/Container.ts

```diff
@@ -42,15 +42,16 @@ export class Container {
      *
      * @param name Service name. Optional
      * @param type Service class
-     * @param params Parameters to be sent to constructor on service initialization
+     * @param params Parameters to be sent to a class constructor or factory function on service initialization
+     * @param factory Factory function to be called on service initialization
      */
-    static registerService(name: string, type: Function, params?: any[]) {
-        this.registeredServices.push({ name: name, type: type, params: params });
+    static registerService(name: string, type: Function, params?: any[], factory?: Function) {
```

Comment:
Also, the return type should be `Object` not `Function`, because factory function returns instances, not functions. So the type of the factory function should be: `(...params: any[]) => Object`. Do you agree?